code/edge/src/main/java/com/adobe/marketing/mobile/CompletionCallbacksManager.java

```diff
@@ -91,7 +93,21 @@ void unregisterCallback(final String requestEventId) {
 
 		if (callback != null) {
 			final List<EdgeEventHandle> handles = edgeEventHandles.get(requestEventId);
-			callback.onComplete(handles != null ? handles : new ArrayList<>());
+			try {
+				callback.onComplete(handles != null ? handles : new ArrayList<>());
+			} catch (Exception ex) {
+				StringWriter stringWriter = new StringWriter();
+				PrintWriter printWriter = new PrintWriter(stringWriter);
+				ex.printStackTrace(printWriter);
+
+				Log.warning(
+					LOG_TAG,
+					LOG_SOURCE,
+					"Exception thrown when invoking EdgeCallback.onComplete() for request event id %s: %s",
+					requestEventId,
+					stringWriter.toString()
```

Comment:
Reading more about this it looks like it is recommended to use the platform level logging support for exceptions instead of printStackTrace. 

@praveek any chance would could support for logging throwable in the Logging service as the platform provides?
https://developer.android.com/reference/android/util/Log#w(java.lang.String,%20java.lang.Throwable)

Alternatively we can look into using https://developer.android.com/reference/android/util/Log#getStackTraceString(java.lang.Throwable)

Reply:
@emdobrin We can explore about adding these additional log methods after the 2.0 release. For now, I would prefer [getStackTraceString](https://developer.android.com/reference/android/util/Log#getStackTraceString(java.lang.Throwable)) to get the string representation of stack trace. 

nit: Can we reword the log to `Exception thrown when invoking completion callback for request event id`
EdgeCallback.onComplete() is not very intuitive confusing when customer passes a lambda.

Reply:
@kevinlind can you make this update and also create a feature request for the logging service in aepsdk-core-android?